File: sdkwork-backend-sdk-java/src/main/java/com/sdkwork/backend/api/VisitHistoryApi.java
package com.sdkwork.backend.api;

import com.sdkwork.backend.http.HttpClient;
import com.sdkwork.backend.model.*;
import java.util.List;
import java.util.Map;

public class VisitHistoryApi {
    private final HttpClient client;
    
    public VisitHistoryApi(HttpClient client) {
        this.client = client;
    }

    /** Update Visit History */
    public PlusApiResultPlusVisitHistoryVO update(PlusVisitHistoryForm body) throws Exception {
        return (PlusApiResultPlusVisitHistoryVO) client.put(ApiPaths.backendPath("/visit_history"), body);
    }

    /** Create Visit History */
    public PlusApiResultPlusVisitHistoryVO create(PlusVisitHistoryForm body) throws Exception {
        return (PlusApiResultPlusVisitHistoryVO) client.post(ApiPaths.backendPath("/visit_history"), body);
    }

    /** List Visit Histories by Page */
    public PlusApiResultPagePlusVisitHistoryVO listByPage(QueryListForm body, Map<String, Object> params) throws Exception {
        return (PlusApiResultPagePlusVisitHistoryVO) client.post(ApiPaths.backendPath("/visit_history/list"), body, params);
    }

    /** List All Visit Histories */
    public PlusApiResultListPlusVisitHistoryVO listAllEntities(QueryListForm body) throws Exception {
        return (PlusApiResultListPlusVisitHistoryVO) client.post(ApiPaths.backendPath("/visit_history/list/all"), body);
    }

    /** Get Visit History by ID */
    public PlusApiResultPlusVisitHistoryVO getById(String id) throws Exception {
        return (PlusApiResultPlusVisitHistoryVO) client.get(ApiPaths.backendPath("/visit_history/" + id + ""));
    }

    /** Delete Visit History */
    public PlusApiResultBoolean delete(String id) throws Exception {
        return (PlusApiResultBoolean) client.delete(ApiPaths.backendPath("/visit_history/" + id + ""));
    }
}

File: sdkwork-backend-sdk-kotlin/src/main/kotlin/com/sdkwork/backend/api/VisitHistoryApi.kt
package com.sdkwork.backend.api

import com.sdkwork.backend.*
import com.sdkwork.backend.http.HttpClient

class VisitHistoryApi(private val client: HttpClient) {

    /** Update Visit History */
    suspend fun update(body: PlusVisitHistoryForm): PlusApiResultPlusVisitHistoryVO? {
        return client.put(ApiPaths.backendPath("/visit_history"), body) as? PlusApiResultPlusVisitHistoryVO
    }

    /** Create Visit History */
    suspend fun create(body: PlusVisitHistoryForm): PlusApiResultPlusVisitHistoryVO? {
        return client.post(ApiPaths.backendPath("/visit_history"), body) as? PlusApiResultPlusVisitHistoryVO
    }

    /** List Visit Histories by Page */
    suspend fun listByPage(body: QueryListForm? = null, params: Map<String, Any>? = null): PlusApiResultPagePlusVisitHistoryVO? {
        return client.post(ApiPaths.backendPath("/visit_history/list"), body, params) as? PlusApiResultPagePlusVisitHistoryVO
    }

    /** List All Visit Histories */
    suspend fun listAllEntities(body: QueryListForm? = null): PlusApiResultListPlusVisitHistoryVO? {
        return client.post(ApiPaths.backendPath("/visit_history/list/all"), body) as? PlusApiResultListPlusVisitHistoryVO
    }

    /** Get Visit History by ID */
    suspend fun getById(id: String): PlusApiResultPlusVisitHistoryVO? {
        return client.get(ApiPaths.backendPath("/visit_history/$id")) as? PlusApiResultPlusVisitHistoryVO
    }

    /** Delete Visit History */
    suspend fun delete(id: String): PlusApiResultBoolean? {
        return client.delete(ApiPaths.backendPath("/visit_history/$id")) as? PlusApiResultBoolean
    }
}
